crates/host/src/wasmbus/mod.rs

```diff
@@ -1431,4 +1606,151 @@ impl Lattice {
+
+        // TODO: Broadcast `linkdef_added`
+
+        let msgp = rmp_serde::to_vec(ld).context("failed to encode link definition")?;
+        self.nats
+            .publish(
+                // TODO: Set prefix
+                format!(
+                    "wasmbus.rpc.{prefix}.{provider_id}.{link_name}.linkdefs.put",
+                    prefix = "default"
+                ),
+                msgp.into(),
+            )
+            .await
+            .context("failed to publish link definition")?;
+        Ok(())
+    }
+
+    #[instrument(skip(self, id, _value))]
+    async fn process_linkdef_delete(
+        &self,
+        id: impl AsRef<str>,
+        _value: impl AsRef<[u8]>,
+    ) -> anyhow::Result<()> {
+        let id = id.as_ref();
+
+        debug!(id, "process link definition entry deletion");
+
+        let mut links = self.links.write().await;
+        // NOTE: There is a race condition here, which occurs when `linkdefs.del`
+        // is used before `data_watch` task has fully imported the current lattice,
+        // but that command is deprecated, so assume it's fine
+        let ref ld @ LinkDefinition {
+            ref actor_id,
+            ref provider_id,
+            ref link_name,
+            ref contract_id,
+            ref values,
+            ..
+        } = links
+            .remove(id)
+            .context("attempt to remove a non-existent link")?;
+
+        self.publish_event(
+            "linkdef_deleted",
+            event::linkdef_deleted(id, actor_id, provider_id, link_name, contract_id, values),
+        )
+        .await?;
+
+        // TODO: Broadcast `linkdef_removed`
+
+        let msgp = rmp_serde::to_vec(ld).context("failed to encode link definition")?;
+        self.nats
+            .publish(
+                // TODO: Set prefix
+                format!(
+                    "wasmbus.rpc.{prefix}.{provider_id}.{link_name}.linkdefs.del",
+                    prefix = "default"
+                ),
+                msgp.into(),
+            )
+            .await
+            .context("failed to publish link definition deletion")?;
+        Ok(())
+    }
+
+    #[instrument(skip(self))]
+    async fn process_entry(
+        &self,
+        async_nats::jetstream::kv::Entry {
+            bucket,
+            key,
+            value,
+            revision,
+            delta,
+            created,
+            operation,
+        }: async_nats::jetstream::kv::Entry,
+    ) {
+        use async_nats::jetstream::kv::Operation;
+
+        let mut key_parts = key.split('_');
+        #[allow(clippy::single_match_else)]
+        let res = match (operation, key_parts.next(), key_parts.next()) {
+            (Operation::Put, Some("LINKDEF"), Some(id)) => {
+                self.process_linkdef_put(id, value).await
+            }
+            (Operation::Delete, Some("LINKDEF"), Some(id)) => {
+                self.process_linkdef_delete(id, value).await
+            }
+            _ => {
```

Comment:
You'll want a TODO here for operations to put/delete `CLAIMS` and `REFMAP` entries as well